### ISSUE TITLE
Have you tried normal back propagation with Internal Feedback?

### ISSUE THREAD

**Issue by universuen** [sunblaze-ucb/Intuitor#5]
Hey guys, excellent work for the community! 

I have a small concern regarding this approach. Since every component of the optimization objective is differentiable, have you considered treating it as a loss function and using the standard training pipeline instead of reinforcement learning? For example:

```
loss = beta * kl_div(model_per_token_prob, ref_per_token_prob) - kl_div(uniform_per_token_prob, model_per_token_prob)
loss.backward()
optimizer.step()
```

Compared to GRPO, this approach may be more efficient.
I would really like to know if there are any early experiment results on this : )
